### PR TITLE
Update structs to schema

### PR DESCRIPTION
**Pull request by BrodyGimson**
Backend ran into problems where our JSON field names deviated from our API spec. This will update all field names for structs and tests to match the API spec. Please have a detailed look especially at the structs.go file to see if I missed anything!

Algorithm team, do not merge until a backend team member has given their approval.

### CODE REVIEW

**Review by treavorgagne — APPROVED** [review]

File: structs/structs.go [hunk 1]
```diff
@@ -7,16 +7,16 @@ type Input struct {
...
+	SummerCourses []Course `json:"summerCourses"`
 }
 
 type Assignment struct {
 	StartDate string  `json:"startDate"`
 	EndDate   string  `json:"endDate"`
 	BeginTime string  `json:"beginTime"`
-	EndTime   string  `json:"endtime"`
+	EndTime   string  `json:"endTime"`
 	HoursWeek float32 `json:"hoursWeek"`
```
Comment (aomi) [inline structs/structs.go:20]: The swagger document lists this as `integer`. Not a huge deal on our end (or probs other alg as JavaScript treats them the same) but just noting.

File: structs/structs.go [hunk 2]
```diff
@@ -34,14 +34,14 @@ type Course struct {
 	CourseTitle       string     `json:"courseTitle"`
 	RequiredEquipment []string   `json:"requiredEquipment,omitempty"`
 	StreamSequence    string     `json:"streamSequence,omitempty"`
-	Assignment        Assignment `json:"meetingTime"`
+	Assignment        Assignment `json:"assignment"`
```
Comment (aomi) [inline structs/structs.go:37]: Is this supposed to be an array?

File: structs/structs.go [hunk 3]
```diff
@@ -34,14 +34,14 @@ type Course struct {
 	CourseTitle       string     `json:"courseTitle"`
 	RequiredEquipment []string   `json:"requiredEquipment,omitempty"`
```
Comment (aomi) [inline structs/structs.go:35]: Maybe this can be deleted as it's not part of anything iirc anymore

Comment (BrodyGimson) [reply structs/structs.go:35]: Yeah I was thinking the same thing as we haven't ever used it

File: structs/structs.go (hunk 2 above)
Comment (BrodyGimson) [reply structs/structs.go:37]: No assignment should just be an object

File: structs/structs.go (hunk 1 above)
Comment (BrodyGimson) [reply structs/structs.go:20]: We were working with strings for the beginTime and endTime as then we can have the leading 0 for times like 0830 etc. for formatting

Comment (aomi) [reply structs/structs.go:20]: I'm talking about `HoursWeek`. Don't think I highlighted the right field.

**Review by treavorgagne — APPROVED** [review]

File: structs/structs.go (hunk 2 above)
Comment (BrodyGimson) [reply structs/structs.go]: Removed required equipment fields

File: structs/structs.go (hunk 3 above)
Comment (BrodyGimson) [reply structs/structs.go]: Removed them

**Review by treavorgagne — APPROVED** [review]

**Review by LacroixJ — APPROVED** [review]
LGTM

**Pull request merged**